docs/api.rst

```diff
@@ -463,6 +463,50 @@ section generates a new *URL* instance.
       >>> URL('http://example.com/path?a=b&b=1').with_query([('b', '2')])
       URL('http://example.com/path?b=2')
 
+.. method:: URL.update_query(query)
+            URL.update_query(**kwargs)
+
+   Returns a new URL with *query* part updated.
```

Comment:
The difference with `with_query` should be highlighted here.
Doc for `with_query` also should mention `update_query` with words like "for replacing only particular query keys use update_query"